README.md

```diff
@@ -38,6 +40,86 @@ hop subscribe kafka://hostname:port/gcn -s EARLIEST
 This will read messages from the gcn topic from the earliest offset
 and read messages until an end of stream (EOS) is received.
 
+#### Python API
+
+Publish messages:
+
+Using the python API, we can publish various types of messages, including
+structured messages such as GCN Circulars and VOEvents:
+
+```python
+    from hop import stream
+    from hop.models import GCNCircular
+
+    # read in a GCN circular
+    with open("path/to/circular.gcn3", "r") as f:
+        circular = GCNCircular.load(f)
+
+    with stream.open("kafka://hostname:port/topic", "w") as s:
+        s.write(circular)
+```
+
+In addition, we can also publish unstructured messages as long as they are
+JSON serializable:
+
+```python
+    from hop import stream
+
+    with stream.open("kafka://hostname:port/topic", "w") as s:
+        s.write({"my": "message"})
+```
+
+By default, authentication is enabled for the Hop broker. In order to authenticate, one
+can pass in an `Auth` instance with credentials:
+
+```python
+    from hop import stream
+    from hop.auth import Auth
+
+    auth = Auth("my-username", "my-password")
+
+    with stream.open("kafka://hostname:port/topic", "w", auth=auth) as s:
+        s.write({"my": "message"})
+```
+
+A convenience function is also provided to read in auth configuration in the same way
+as in the CLI client:
+
+```python
+    from hop import stream
+    from hop.auth import load_auth
+
+    with stream.open("kafka://hostname:port/topic", "w", auth=load_auth()) as s:
+        s.write({"my": "message"})
+```
+
+Consume messages:
+
+
+```python
+    from hop import stream
+
+    with stream.open("kafka://hostname:port/topic", "r") as s:
+        for message in s:
+             print(message)
+```
+
+This will listen to the Hop broker, listening to new messages and printing them to
+stdout as they arrive until there are no more messages in the stream.
```

Comment:
I think the stream will still be open even if there are no more messages (open forever), right? if so, I guess we need to rephrase this sentence.

Reply:
No, it'll close now by default when it reads all the messages in the stream thanks to the awesome work Spencer did in adc-streaming 1.x. To listen to messages forever, now we need the `persist=True` flag in the open connection.

Reply:
okay, it's clearer now to me. Thanks.